azure-kusto-data/azure/kusto/data/helpers.py

```diff
@@ -49,32 +78,25 @@ def dataframe_from_result_table(table: "Union[KustoResultTable, KustoStreamingRe
     columns = [col.column_name for col in table.columns]
     frame = pd.DataFrame(table.raw_rows, columns=columns)
 
-    # fix types
+    # converters_by_type overrides the default
+    if converters_by_type and not nullable_bools:
+        converters_by_type = {**default_dict, **converters_by_type}
+    elif converters_by_type:
+        converters_by_type = {**default_dict, **{"bool": lambda col, df: df[col].astype(pd.BooleanDtype())}, **converters_by_type}
+    else:
+        converters_by_type = default_dict
+
     for col in table.columns:
-        if col.column_type == "string" and hasattr(pd, "StringDType"):
-            frame[col.column_name] = frame[col.column_name].astype(pd.StringDType())
-        if col.column_type == "bool":
-            frame[col.column_name] = frame[col.column_name].astype(pd.BooleanDtype() if nullable_bools else bool)
-        elif col.column_type == "int":
-            frame[col.column_name] = frame[col.column_name].astype(pd.Int32Dtype())
-        elif col.column_type == "long":
-            frame[col.column_name] = frame[col.column_name].astype(pd.Int64Dtype())
-        elif col.column_type == "real" or col.column_type == "decimal":
-            frame[col.column_name] = frame[col.column_name].replace("NaN", np.NaN).replace("Infinity", np.PINF).replace("-Infinity", np.NINF)
-            frame[col.column_name] = pd.to_numeric(frame[col.column_name], errors="coerce").astype(pd.Float64Dtype())
-        elif col.column_type == "datetime":
-            # Pandas before version 2 doesn't support the "format" arg
-            args = {}
-            if pd.__version__.startswith("2."):
-                args = {"format": "ISO8601", "utc": True}
+        if converters_by_column_name and col.column_name in converters_by_column_name.keys():
+            if isinstance(converters_by_column_name[col.column_name], types.LambdaType):
```

Comment:
This is flimsy, if the users provide a function for example it won't do as expected. On the other hand, I don't think there is a common base for python types and pandas types, 

So with that I think we should remove the "giving just a type" functionality and only keep the converter lambdas

Reply:
@ohadbitton maybe you have an idea

Reply:
any concern in just checking if its a string type instead?

Reply:
Seems like a good compromise. 

Make sure of course to change the type of the converters to `Dict[str, Union[str, Callable[[str, "pd.DataFrame"], "pd.Series"]]]`